app/services/meal/create.rb

```diff
@@ -0,0 +1,34 @@
+class Meal
+  class Create < ApplicationService
+    def initialize(params = {})
+      @meal_parmas = params[:params][:meal]
+      @form =        params[:form]
+      @menu =        params[:menu]
+    end
+
+    def call
+      save_meal
+    end
+
+    private
+
+    attr_reader :form, :meal_parmas, :menu
+
+    def save_meal
+      return unless form.validate(meal_parmas)
+      form.save do |valid_params|
+        add_to_menu(valid_params)
+      end
+    end
+
+    def add_to_menu(valid_params)
+      meal = meal(valid_params)
+
+      menu.meals << meal unless menu.meals.include? meal
```

Comment:
Use delegation more, e.g `delegate :meals, to: :menu, prefix: true #=> menu_meals.include?(meal)`